nelox/interpreter.py

```diff
@@ -0,0 +1,223 @@
+from nelox.Expr import Literal, Variable, List
+
+
+class Environment:
+    def __init__(self, parent=None):
+        self.values = {}
+        self.parent = parent
+
+    def get(self, name):
+        if name in self.values:
+            return self.values[name]
+        elif self.parent:
+            return self.parent.get(name)
+        else:
+            raise RuntimeError(f"Undefined variable '{name}'")
+
+    def set(self, name, value):
+        if name in self.values:
+            self.values[name] = value
+        elif self.parent:
+            try:
+                self.parent.set(name, value)
+            except RuntimeError:
+                self.values[name] = value
```

Comment:
This is a very general catch - it is generally a bad idea to hide all errors like this.  (Besides, what is it that can fail?  I don't see any raises here.)